aidial_adapter_vertexai/chat/static_tools.py

```diff
@@ -0,0 +1,82 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import List, Self
+
+from aidial_sdk.chat_completion.request import (
+    AzureChatCompletionRequest,
+    StaticTool,
+)
+from pydantic import BaseModel
+from vertexai.preview.generative_models import Tool as GeminiTool
+from vertexai.preview.generative_models import grounding
+
+from aidial_adapter_vertexai.chat.errors import ValidationError
+
+
+class ToolName(str, Enum):
+    """
+    https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/grounding
+    """
+
+    GOOGLE_SEARCH = "google_search"
+
+
+class StaticToolProcessor(ABC):
+    @abstractmethod
+    def validate_config(self, config: dict | None) -> None: ...
+
+    @abstractmethod
+    def to_gemini_tools(self, tool: StaticTool) -> List[GeminiTool]: ...
+
+    def process(self, tool: StaticTool) -> List[GeminiTool]:
+        self.validate_config(tool.static_function.configuration)
+        return self.to_gemini_tools(tool)
+
+
+class GoogleSearchGroundingTool(StaticToolProcessor):
+    def validate_config(self, config: dict | None) -> None:
+        if config:
+            raise ValidationError(
+                "Google search tool doesn't support configuration"
+            )
+
+    def to_gemini_tools(self, tool: StaticTool) -> List[GeminiTool]:
+        return [
```

Comment:
It doesn't make sense to separate `StaticTool` validation from `StaticTool` to `GeminiTool` parsing/conversion since they depend on each other.

You only need `to_gemini_tools` method in `StaticToolProcessor` class.
And even further, you could make processors **composable**:

```diff
- def to_gemini_tools(self, tool: StaticTool) -> List[GeminiTool]: ...
+ @staticmethod
+ def parse_gemini_tools(tool: StaticTool) -> Optional[List[GeminiTool]]: ... 
```

Then parsing will be as easy as this:

```python
tools = GoogleSearchGroundingTool.parse_gemini_tools(tool) or WhatEverTool.parse_gemini_tools(tool) or unknown_tool(tool)

def unknown_tool_name(tool):
    throw ValidationError(f"Unsupported static tool: {tool.static_function.name}")
```